spec/rails_app/config/application.rb

```diff
@@ -5,11 +5,34 @@ class Application < Rails::Application
     config.root = File.expand_path("../../", __FILE__)
     config.i18n.enforce_available_locales = true
 
-    if !Rails.version.start_with?("5.0") && !Rails.version.start_with?("5.1") && config.active_record.respond_to?(:yaml_column_permitted_classes=)
+    if Rails.version.start_with?("7.1") && config.active_record.respond_to?(:yaml_column_permitted_classes=)
+      config.active_record.yaml_column_permitted_classes = [
+        String,
+        Symbol,
+        Integer,
+        NilClass,
+        Float,
+        Time,
+        Date,
+        FalseClass,
+        Hash,
+        Array,
+        DateTime,
+        TrueClass,
+        BigDecimal,
+        ActiveSupport::TimeWithZone,
+        ActiveSupport::TimeZone,
+        ActiveSupport::HashWithIndifferentAccess
+      ]
```

Comment:
Apparently the elements of permitted classes need to be actual class objects rather than strings. Otherwise there would be a nuch of failing tests with the error message below:

```
  19) Audited::Auditor revisions should be able to get time for first revision
      Failure/Error: ActiveRecord::Coders::YAMLColumn.new(Object).dump(obj)

      Psych::DisallowedClass:
        Tried to dump unspecified class: ActiveSupport::TimeWithZone
      # ./lib/audited/audit.rb:30:in `dump'
      # ./lib/audited/auditor.rb:363:in `block in write_audit'
      # ./lib/audited/auditor.rb:362:in `write_audit'
      # ./lib/audited/auditor.rb:331:in `audit_create'
      # ./spec/audited/auditor_spec.rb:820:in `block (3 levels) in <top (required)>'
```

I have taken a look at [Psych's code](https://github.com/ruby/psych/commit/441958396f40d526a62e564761d2bad534465a5b#diff-c31f86c71837c4f06859aeb1a7ef9e17f42420bae72e22c6eade6ef8b4adc7fbR418),  and it does seem like that this has always been class objects rather than strings. Not sure where the original setting has derived, but either way, this should fix the test for Rails 7.1.